src/lib/optimizer/optimizer.cpp

```diff
@@ -59,6 +61,11 @@ std::shared_ptr<Optimizer> Optimizer::create_default_optimizer() {
 
   optimizer->add_rule(std::make_unique<PredicateSplitUpRule>());
 
+  // Re-run ExpressionReductionRule to optimize LIKE's that have been part of a disjunction expression evaluation
+  // (e.g., `col_a LIKE 'a%' OR col_a LIKE 'b%'; cf. JOB's query 19b; split up by PredicateSplitUpRule). This time,
+  // also add between predicates.
+  optimizer->add_rule(std::make_unique<ExpressionReductionRule>(true));
+
   optimizer->add_rule(std::make_unique<NullScanRemovalRule>());
```

Comment:
Since we re-run all the other expression reduction code: Does it make sense to put the LIKE transformations into a separate rule and call it here, or do we profit from re-running

* remove_duplicate_aggregate
* reduce_distributivity
* reduce_constant_expression

as well ?

Reply:
Very good point. I'll need to think about it, but I think you're right. Makes more sense with a new rule.